lib/vm/src/instance/ref.rs

```diff
@@ -208,3 +210,13 @@ impl Drop for InstanceRef {
         unsafe { Self::deallocate_instance(self) };
     }
 }
+
+impl MemoryUsage for InstanceRef {
+    fn size_of_val(&self, tracker: &mut dyn MemoryUsageTracker) -> usize {
+        mem::size_of_val(self) + self.strong.size_of_val(tracker) - mem::size_of_val(&self.strong)
+            + self.instance_layout.size_of_val(tracker)
+            - mem::size_of_val(&self.instance_layout)
+            + self.as_ref().size_of_val(tracker)
+            - mem::size_of_val(&self.instance)
```

Comment:
Is it possible to encapsulate the semantics of this into the loupe library itself more directly? It just looks very confusing without context

Reply:
I will try to come with an API for that.